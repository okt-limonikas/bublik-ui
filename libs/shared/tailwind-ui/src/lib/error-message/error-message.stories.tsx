/* SPDX-License-Identifier: Apache-2.0 */
/* SPDX-FileCopyrightText: 2021-2023 OKTET Labs Ltd. */
import { StoryFn, Meta } from '@storybook/react';
import { ErrorMessage, ErrorMessageProps } from './error-message';

export default {
	component: ErrorMessage,
	title: 'form/Error Message',
	decorators: [
		(Story) => <div className="p-4 bg-white rounded-md">{Story()}</div>
	]
} as Meta;

const Template: StoryFn<ErrorMessageProps> = (args) => (
	<ErrorMessage>Test Label</ErrorMessage>
);

export const Primary = {
	render: Template,
	args: {}
};
