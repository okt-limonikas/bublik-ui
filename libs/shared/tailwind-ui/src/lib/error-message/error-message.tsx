/* SPDX-License-Identifier: Apache-2.0 */
/* SPDX-FileCopyrightText: 2021-2023 OKTET Labs Ltd. */
import { ComponentPropsWithRef, forwardRef } from 'react';

import { cn } from '../utils';

export type ErrorMessageProps = ComponentPropsWithRef<'span'>;

export const ErrorMessage = forwardRef<HTMLSpanElement, ErrorMessageProps>(
	({ children, className, ...props }, ref) => {
		return (
			<span
				className={cn('text-[0.75rem] font-normal text-bg-error', className)}
				{...props}
				ref={ref}
				data-testid="input-error-message"
			>
				{children}
			</span>
		);
	}
);
