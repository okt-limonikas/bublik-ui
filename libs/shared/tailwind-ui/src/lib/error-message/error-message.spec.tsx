/* SPDX-License-Identifier: Apache-2.0 */
/* SPDX-FileCopyrightText: 2021-2023 OKTET Labs Ltd. */
import { it, describe, expect } from 'vitest';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

import { ErrorMessage } from './error-message';

describe('ErrorMessage', () => {
	it('should render successfully', () => {
		const { getByTestId } = render(<ErrorMessage />);

		const element = getByTestId('input-error-message');

		expect(element).toBeVisible();
	});

	it('should match snapshot', () => {
		const tree = renderer.create(<ErrorMessage />).toJSON();

		expect(tree).toMatchSnapshot();
	});
});
