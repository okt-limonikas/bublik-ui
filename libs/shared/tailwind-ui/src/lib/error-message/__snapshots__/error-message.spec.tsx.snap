// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ErrorMessage > should match snapshot 1`] = `
<span
  className="text-[0.75rem] font-normal text-bg-error"
  data-testid="input-error-message"
/>
`;
